src/diffusers/models/model_loading_utils.py

```diff
@@ -204,6 +205,23 @@ def _fetch_index_file(
             subfolder or "",
             _add_variant(SAFE_WEIGHTS_INDEX_NAME if use_safetensors else WEIGHTS_INDEX_NAME, variant),
         )
+        if not os.path.exists(index_file) and variant is not None:
```

Comment:
what happends if both formats exists? I think we can keep this function as it is, and only if we cannot find the index with the format, we try to fetch the legacy format (put this logic in a separate function )

Reply:
https://github.com/huggingface/diffusers/pull/9253#discussion_r1758081569

Reply:
This is needed for loading from the model level, no?
This we have been supporting https://github.com/huggingface/diffusers/pull/9061